src/plugins/dashboard/public/application/dashboard_app_no_data.tsx

```diff
@@ -10,37 +10,45 @@ import React from 'react';
 import {
   AnalyticsNoDataPageKibanaProvider,
   AnalyticsNoDataPage,
+  AnalyticsNoDataPageKibanaDependencies,
 } from '@kbn/shared-ux-page-analytics-no-data';
-import { DataPublicPluginStart } from '@kbn/data-plugin/public';
 
-import { DashboardAppServices } from '../types';
-import { useKibana } from '../services/kibana_react';
+import { pluginServices } from '../services/plugin_services';
 
 export const DashboardAppNoDataPage = ({
   onDataViewCreated,
 }: {
   onDataViewCreated: () => void;
 }) => {
   const {
-    services: { core, data, dataViewEditor },
-  } = useKibana<DashboardAppServices>();
+    application,
+    data: { dataViews },
+    dataViewEditor,
+    http: { basePath },
+    documentationLinks: { kibanaGuideDocLink },
+  } = pluginServices.getServices();
+
   const analyticsServices = {
-    coreStart: core as unknown as React.ComponentProps<
-      typeof AnalyticsNoDataPageKibanaProvider
-    >['coreStart'],
-    dataViews: data.dataViews,
+    coreStart: {
+      docLinks: { links: { kibana: { guide: kibanaGuideDocLink } } },
+      application,
+      http: { basePath },
+    },
+    dataViews,
     dataViewEditor,
-  };
+  } as unknown as AnalyticsNoDataPageKibanaDependencies;
```

Comment:
You calling this out made me look closer at it, and seems like I was missing one small part of the `AnalyticsNoDataPageKibanaDependencies` - specifically, `coreStart.docLinks.indexPatterns.introduction`. 

I tested the no data page so I'm **pretty sure** it all worked, but possible I missed clicking a specific link or something that would have been broken because of this? Either way, I added that field to the `documentationLinksService` and the typecasting as been removed as of a82f3aa55bd833a140b898f9e5a31db6acb0bebe 👍